bcda/alr/csv/csv.go

```diff
@@ -0,0 +1,144 @@
+package csv
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/CMSgov/bcda-app/bcda/models"
+	"github.com/dimchansky/utfbom"
+	"github.com/go-gota/gota/dataframe"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	mbi   = "BENE_MBI_ID"
+	hic   = "BENE_HIC_NUM"
+	first = "BENE_1ST_NAME"
+	last  = "BENE_LAST_NAME"
+	sex   = "BENE_SEX_CD"
+	birth = "BENE_BRTH_DT"
+	death = "BENE_DEATH_DT"
+)
+
+// Fields the must be present in an ALR file
+var requiredFields []string = []string{
+	mbi, hic,
+	first, last,
+	sex,
+	birth, death,
+}
+
+// Fields that will be used to join multiple dataframes together
+var joinFields = requiredFields
+
+// ToALR reads in a CSV file(s) and unmarshals the data into an ALR model.
+// CSV files are joined based on a predetermined list of fields
+func ToALR(names ...string) ([]models.Alr, error) {
+	var mergedDF dataframe.DataFrame
+	for _, name := range names {
+		df, err := toDataFrame(name)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create dataframe: %w", err)
+		}
+		if err := validate(df); err != nil {
```

Comment:
Should we talk to Bill or Nicole about what to do when a particular ALR table is missing the required fields. Do we just not process the whole ALR submission or just skip that particular table.

Though if one table is sparse, probably a good chance other tables are sparse too. I lean towards former, and not process the whole submission like you have here.

Reply:
Thinking the former as well. I'm thinking if an ALR violates some data assumptions, we should skip this conversion to ensure that we have some guarantees on the data that we're going to serve. We'd be able to say that we will have Table 1-1, 1-2, etc. Instead of saying that we may have situations where we have 1-1, 1-3, 1-4. 1-2 had some bad data.

We should keep in mind a "best-effort" approach too, but I think that should be weighted w/ Nicole, Laureen, and H-Team if we decide to pivot to that approach.